core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -423,4 +435,76 @@ public void run() {
       }
     }
   }
+
+  public static final Metadata.Key<Long> TIMEOUT_KEY =
```

Comment:
Uhhh... good question. Not quite sure :-)

I think we don't have very many choices: 1) we can add a `getDeadline()` to `ServerCall`. 2) we could expose `TIMEOUT_KEY` and just force accessing it from `Metadata`. `ServerImpl` will need to access it from `Metadata` in order to set up timers (assuming that happens in #58) and configure Context in #262. I expect (1) may be preferred, but not much discussion has happened.